### PR TITLE
Change blog to sidebar layout

### LINKED ISSUE #2824: Implement new layout for blog page
The [new blog design](https://github.com/18F/design-lab/issues/17#issuecomment-402828003) ...
- [x] requires a sidebar layout instead of the existing full-page one.
- [x] moves the search bar out of the header into the sidebar.
- [x] adds new lines of content to the header.
- [x] adds a footer with a "get in touch" call to action.

### PR DESCRIPTION
**Pull request by hbillings**
Closes #2824.

Currently looks like:
screencapture-0-0-0-0-4000-site-blog-2018-07-24-18_11_56

Outstanding items/questions:
- The new header text for the blog page lives in the `blog_tagline` frontmatter variable in `blog/index.html`. As soon as we've got this copy, we can drop it into that variable!
- Blog footer text is also a frontmatter variable in the same page.
- Wasn't sure what to do with the RSS feed, so I kept it under the search bar as it currently is. 

Also note I'm OOO until 8/5, so will circle back around then.

### CODE REVIEW

File: _sass/_components/blog-index.scss [hunk 1]
```diff
@@ -1,25 +1,42 @@
...
-        display: initial;
-      }
-    }
+.blog-footer {
+  h2 {
+    font-size: $base-font-size;
+    font-weight: normal;
+  }
+
+  a.usa-button {
```
[the remark this answers is not in the document]
Comment (hbillings) [reply _sass/_components/blog-index.scss]: yes well thanks

**Comment by eddietejeda** [comment]
great progress!

File: blog/index.html [hunk 2]
```diff
@@ -2,15 +2,33 @@
 layout: default
 title: Blog
```
Comment (awfrancisco) [inline blog/index.html]: Change to `18F Blog`

File: blog/index.html [hunk 3]
```diff
@@ -2,15 +2,33 @@
...
 skip_index: true
+blog_tagline: A view of how we deliver civic technology
```
Comment (awfrancisco) [inline blog/index.html]: Change to `Delivering civic technology`

**Comment by hbillings** [comment]
Obviated by #2890.

**Pull request closed without merging**